src/app/modules/auth/services/regobs-auth.service.ts

```diff
@@ -79,13 +77,14 @@ export class RegobsAuthService {
 
     // TODO: This could be saved in offline storage if we want to presist
     // myPage data and the user start app when offline and want to edit registration
-    this.myPageData$ = this.loggedInUser$.pipe(
-      switchMap((loggedInUser) => {
-        if (loggedInUser.isLoggedIn) {
-          return this.accountApi.AccountGetMyPageData();
-        } else {
-          return of(undefined);
-        }
+    this.myPageData$ = merge(this.loggedInUser$, this.refreshMyPageData$).pipe(
```

Comment:
Her kan du også vurdere denne metoden:

```
combineLatest([this.loggedInUser$, this.refreshMyPageData$.pipe(startsWith(null))]).pipe(......) 
```

hvis du vil slippe å subscribe til loggedInUser på nytt. Jeg syns kanskje det er litt ryddigere.

Grunnen til at du i så fall trenger startsWith(..) på refreshMyPageData er for å være sikker på at combineLatest starter som den skal. Den emiter bare når begge observables har emitet.

Reply:
hvis jeg bruker combineLatest klarer jeg ikke å få noe data. myPage er undefined